### PR TITLE
Update docs and tests for struct literals

### PR DESCRIPTION
**Pull request by ntrel**
* Update tests for struct literal function arguments
* Update docs: struct literals without struct name

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -710,21 +710,30 @@ struct Point {
...
+// Alternative literal syntax for structs with 3 fields or fewer
+p = Point{10, 20}
+assert p.x == 10
+
+// you can omit the struct name when it's already known
+p = {x: 30, y: 4}
+assert p.y == 4
 ```
 
+Omitting the struct name also works for function arguments.
```
Comment (medvednikov) [inline doc/docs.md:729]: This is covered in https://github.com/vlang/v/blob/master/doc/docs.md#short-struct-initialization-syntax

I don't think it needs to be duplicated.

Comment (ntrel) [reply doc/docs.md:729]: Actually that section doesn't mention `func({struct_field: init})`.
I've just added a commit here that tweaks that section.

File: doc/docs.md [hunk 2]
```diff
@@ -802,7 +814,7 @@ instead of
 new_button(ButtonConfig{text:'Click me', width:100})
 ```
 
-This only works with functions that have a single struct argument.
+This only works for functions that have a struct for the last argument.
```
Comment (medvednikov) [inline doc/docs.md:817]: no, not last, there needs to be just one struct arg

Comment (ntrel) [reply doc/docs.md:817]: Preceding arguments do work:
https://github.com/vlang/v/pull/5726/files#diff-be137ec37ed574953500274b6e369f46R253

Is this a compiler bug? I think it's useful behaviour.

Comment (medvednikov) [reply doc/docs.md:817]: Hm indeed it does. Strange. I'll need to think about it.

File: doc/docs.md (hunk 1 above)
Comment (ntrel) [reply doc/docs.md:729]: I decided it would be better to put all the short syntax code in one section - see #5762.

**Pull request merged**